web/src/narrow.js

```diff
@@ -125,14 +124,14 @@ export function activate(raw_terms, opts) {
         raw_terms = [{operator: "is", operand: "home"}];
     }
     const filter = new Filter(raw_terms);
+    const filter_to_all_messages_view = filter.is_in_home();
 
-    const is_narrowed_to_all_messages_view = narrow_state.filter()?.is_in_home();
-    if (has_visited_all_messages && is_narrowed_to_all_messages_view && filter.is_in_home()) {
+    if (filter_to_all_messages_view && message_lists.current?.data.filter.is_in_home()) {
         // If we're already looking at the All messages view, exit without doing any work.
         return;
     }
```

Comment:
This is fine for now since it's not a regression, but in the future we may want to consider whether this check is correct -- since there are parameters around the scroll position that we perhaps should be considering.